### PR TITLE
Setup GlobalCache: 30397_1 provide a ilDB while collecting GlobalCache metrics

### PR DESCRIPTION
**Pull request by daniwe4**
https://mantis.ilias.de/view.php?id=30397

### CODE REVIEW

File: Services/GlobalCache/classes/Setup/class.ilGlobalCacheMetricsCollectedObjective.php [hunk 1]
```diff
@@ -15,11 +16,21 @@ public function getTentativePreconditions(Setup\Environment $environment) : arra
...
         }
 
+        // ATTENTION: This is a total abomination. It only exists to allow various
+        // sub components of the various readers to run. This is a memento to the
+        // fact, that dependency injection is something we want. Currently, every
+        // component could just service locate the whole world via the global $DIC.
+        $DIC = $GLOBALS["DIC"];
+        $GLOBALS["DIC"] = new DI\Container();
+        $GLOBALS["DIC"]["ilDB"] = $db;
+
```
Comment (chfsx) [inline Services/GlobalCache/classes/Setup/class.ilGlobalCacheMetricsCollectedObjective.php:33]: Do we need to "destroy" this after running this Objective?

File: Services/GlobalCache/classes/Setup/class.ilGlobalCacheMetricsCollectedObjective.php [hunk 2]
```diff
@@ -98,5 +109,7 @@ public function collectFrom(Setup\Environment $environment, Setup\Metrics\Storag
             "component_activation",
             $component_activation
         );
+
+        $GLOBALS["DIC"] = $DIC;
```
Comment (chfsx) [inline Services/GlobalCache/classes/Setup/class.ilGlobalCacheMetricsCollectedObjective.php:113]: ah sorry, I think this does the "destroy" thing, right?

Comment (klees) [reply Services/GlobalCache/classes/Setup/class.ilGlobalCacheMetricsCollectedObjective.php:113]: Yes. We try to restore the environment to the previous state to be as non-invasive as possible =)

**Comment by chfsx** [comment]
Thanks for the fix! I'll CP to trunk as well

**Pull request merged**